src/controllers/userController.ts

```diff
@@ -33,5 +34,38 @@ const createUser = async (req: Request, res: Response): Promise<void> => {
     res.status(500).send('Internal Server Error');
   }
 };
+const updatePassword = async (req: Request, res: Response): Promise<void> => {
+  try {
+    const { oldPassword, newPassword, email } = req.body;
+    const saltRound = await bcrypt.genSalt(10);
+    const user = await User.findOne({
+      where: {
+        email: email,
+      },
+    });
+
+    if (!user) {
+      res.status(400).send('User not found');
+      return;
+    }
+    const match = await bcrypt.compare(oldPassword, user.password);
+    if (!match) {
+      res.status(400).send('The old password is incorrect!');
+    }
+    const hashedNewPassword = await bcrypt.hash(newPassword, saltRound);
+    await User.update(
+      { password: hashedNewPassword },
+      {
+        where: {
+          email: email,
+        },
+      }
+    );
+    res.status(200).json({ message: 'Successfully updated user password!' });
+  } catch (error) {
+    logger.error('Error updating user:', error);
+    res.status(500).send('Internal Server Error');
```

Comment:
Please use the reusable function sendInternalErrorResponse from src/validations. Refer to how other controller functions have utilized it.

Reply:
I changed to using the reusable fucntions